pkg/controller/installation/products/mobiledeveloperconsole/reconciler_test.go

```diff
@@ -0,0 +1,583 @@
+				InstallOperatorFunc: func(ctx context.Context, serverClient pkgclient.Client, owner ownerutil.Owner, os marketplacev1.OperatorSource, t marketplace.Target, operatorGroupNamespaces []string, approvalStrategy operatorsv1alpha1.Approval) error {
+					return errors.New("dummy error")
+				},
+			},
+			FakeClient: fakeclient.NewFakeClient(),
+			FakeConfig: basicConfigMock(),
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.Name, func(t *testing.T) {
+			_, err := NewReconciler(tc.FakeConfig, tc.Installation, tc.FakeMPM)
+			if err != nil && err.Error() != tc.ExpectedError {
+				t.Fatalf("unexpected error : '%v', expected: '%v'", err, tc.ExpectedError)
+			}
+			if err == nil && tc.ExpectedError != "" {
+				t.Fatalf("expected error '%v' and got nil", tc.ExpectedError)
+			}
+		})
+	}
+}
+
+func TestReconciler_reconcileCustomResource(t *testing.T) {
+	scheme, err := getBuildScheme()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	cases := []struct {
+		Name           string
+		FakeClient     pkgclient.Client
+		FakeConfig     *config.ConfigReadWriterMock
+		Installation   *v1alpha1.Installation
+		ExpectError    bool
+		ExpectedStatus v1alpha1.StatusPhase
+		FakeMPM        *marketplace.MarketplaceInterfaceMock
+	}{
+		{
+			Name:           "Test reconcile custom resource returns in progress when successful created",
+			FakeClient:     fakeclient.NewFakeClientWithScheme(scheme, getOauthClientSecret(), getOperatorDC(), getRoute()),
+			FakeConfig:     basicConfigMock(),
+			Installation:   &v1alpha1.Installation{},
+			ExpectedStatus: v1alpha1.PhaseCompleted,
+		},
+		{
+			Name: "Test reconcile custom resource returns failed when cr status is failed",
+			FakeClient: &moqclient.SigsClientInterfaceMock{
+				CreateFunc: func(ctx context.Context, obj runtime.Object) error {
+					return errors.New("dummy create error")
+				},
+				GetFunc: func(ctx context.Context, key types.NamespacedName, obj runtime.Object) error {
+					return errors.New("get request failed")
+				},
+			},
+			FakeConfig:     basicConfigMock(),
+			Installation:   &v1alpha1.Installation{},
+			ExpectedStatus: v1alpha1.PhaseFailed,
+			ExpectError:    true,
+		},
+	}
+	for _, tc := range cases {
+		t.Run(tc.Name, func(t *testing.T) {
+			reconciler, err := NewReconciler(tc.FakeConfig, tc.Installation, tc.FakeMPM)
+			if err != nil {
+				t.Fatal("unexpected err ", err)
+			}
+
+			phase, err := reconciler.reconcileCustomResource(context.TODO(), tc.FakeClient, tc.Installation)
+			if tc.ExpectError && err == nil {
+				t.Fatal("expected an error but got none")
+			}
+			if !tc.ExpectError && err != nil {
+				t.Fatal("expected no error but got one ", err)
+			}
+			if tc.ExpectedStatus != phase {
+				t.Fatal("expected phase ", tc.ExpectedStatus, " but got ", phase)
+			}
+		})
+	}
+}
+
+func TestReconciler_handleProgress(t *testing.T) {
+	scheme, err := getBuildScheme()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	unreadyPods := []runtime.Object{}
+	for i := 0; i < 2; i++ {
```

Comment:
Can perhaps check resource for completed status here instead of checking pods?

Reply:
I'd definitely prefer this if it is possible. Is it @jjaferson?

Reply:
Yes, it's possible and much better indeed